### ISSUE TITLE
snmptrapd problem running embedded perl using external modules

### ISSUE THREAD

**Issue by Andrea77S** [net-snmp/net-snmp#262]
Hello.
I can't figure out what is the root cause of the problem:
I'm running snmptrapd on a CentOS 7.9 Linux system to receive traps and convert them into syslog events, after some filtering.
The script has been running fine on SLES11, but on the new server i have a weird problem:
If i run the snmptrapd -f command on a open shell everything works fine, but if the snmptrapd is run by systemd or if I run it in background (using "&" at the end of the line) it doesn't work.

After some testing, i figured out the problem is loading the external modules i'm using in the perl script:

use Net::Syslog;
use Net::Subnet;

running a test script without external modules works fine.

please help.

**Comment by jridky** [comment]
Hi @Andrea77S 

do you have any information e.g. from journal that could tell us a bit more about why are those imports problematic when used in  program that runs as system service?

Josef

**Comment by Andrea77S** [comment]
Hi Jridky, thank you for your fast reply.
Unfortunately I don't, the systemctl service runs correctly, without any errors. Also if I run it from shell, using the same command line used by systemd, it runs perfectly.

I have made a very quick try using the example from this page: https://metacpan.org/pod/NetSNMP::TrapReceiver
Everything works fine (even using systemd or backgrounding from bash) until i enable any module.

I have attached my test script and the snmptrapd.conf and /etc/sysconfig/snmptrapd configuration files for reference.
[test.pl.txt](https://github.com/net-snmp/net-snmp/files/5845178/test.pl.txt)
[snmptrapd-sysconfig.txt](https://github.com/net-snmp/net-snmp/files/5845181/snmptrapd-sysconfig.txt)
[snmptrapd.conf.txt](https://github.com/net-snmp/net-snmp/files/5845182/snmptrapd.conf.txt)

**Comment by Andrea77S** [comment]
Maybe I have solved the problem, it is not a net-snmp problem, but an environment (mis)configuration.
I have found the solution on this post: https://stackoverflow.com/questions/33636231/installed-cpan-modules-in-problematic-location
It seems that MCPAN changes the .bashrc file for the root user pointing to an installation directory under /root/perl5 instead of using the system paths.

Reinstalled the modules it seems to work fine even under systemd.
